### PR TITLE
add check code quality + citations to nightly

### PR DESCRIPTION
**Pull request by mgovers**
* [x] test run in https://github.com/PowerGridModel/power-grid-model/actions/runs/11104508116

### CODE REVIEW

**Review by TonyXiang8787 — DISMISSED** [review]

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -22,7 +22,13 @@ jobs:
     with:
```
Comment (TonyXiang8787) [inline .github/workflows/nightly.yml:22]: why do we give the permission write here?

Comment (mgovers) [reply .github/workflows/nightly.yml:22]: it is needed by the Release job in `main.yml` (note that the actual release is skipped)

Comment (TonyXiang8787) [reply .github/workflows/nightly.yml:22]: If the release step is skipped. We don't need that permission right?

Comment (mgovers) [reply .github/workflows/nightly.yml:22]: theoretically, we don't, but the full sub-workflow needs the same permissions and that can't depend on the fact that some step is skipped somewhere later in the action. just like how in C++, a functions' `const`-ness can't depend on its arguments:

if you want to remove these permissions, the release job needs to be in a separate workflow

**Review by TonyXiang8787 — APPROVED** [review]

**Comment by mgovers** [comment]
adding auto-merge even though the test run has not fully finished yet, because the jobs that haven't finished yet are things that are not affected by this PR

**Pull request merged**